atlas-core/src/main/java/io/atlasmap/core/AtlasMappingService.java

```diff
@@ -39,208 +43,303 @@
 import java.util.List;
 
 public class AtlasMappingService implements Serializable {
-    
+
     public enum AtlasMappingFormat {
         XML("xml"),
         JSON("json");
-        
+
         private String value;
-        
+
         private AtlasMappingFormat(String value) { this.value = value; }
         public String value() { return value; }
     }
+
+    private static final long serialVersionUID = 1668362984516180517L;
+    private static final Logger logger = LoggerFactory.getLogger(AtlasMappingService.class);
+    private transient JAXBContext ctx = null;
+    private transient Marshaller jaxbMarshaller = null;
+    private transient Unmarshaller jaxbUnmarshaller = null;
+    private transient ObjectMapper jsonMapper = null;
+
+    private static final String CONFIG_V2_PACKAGE = "io.atlasmap.v2";
+
+    public AtlasMappingService() {
+        try {
+            List<String> tmp = new ArrayList<String>();
+            tmp.add(CONFIG_V2_PACKAGE);
+            initialize(tmp);
+        } catch (Exception e) {
+            logger.error("Error initializing JAXB: " + e.getMessage(), e);
+            throw new IllegalStateException(e.getMessage(), e);
+        }
+    }
+
+    public AtlasMappingService(List<String> modulePackages) {
+        try {
+            initialize(modulePackages);
+        } catch (Exception e) {
+            logger.error("Error initializing JAXB: " + e.getMessage(), e);
+            throw new IllegalStateException(e.getMessage(), e);
+        }
+    }
+
+    protected void initialize(List<String> packages) throws JAXBException {
+
+        if (getJAXBContext() == null) {
+            setJAXBContext(JAXBContext.newInstance(stringListToColonSeparated(packages)));
+            if(logger.isDebugEnabled()) {
+                logger.debug("Initialized JAXBContext: " + stringListToColonSeparated(packages));
+            }
+        }
+
+        jaxbMarshaller = getJAXBContext().createMarshaller();
+        jaxbMarshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
+        jaxbUnmarshaller = getJAXBContext().createUnmarshaller();
+        jsonMapper = new AtlasJsonMapper();
+    }
+    
+    public AtlasMapping loadMapping(String fileName) throws AtlasValidationException {
+        return loadMapping(fileName, AtlasMappingFormat.XML);
+    }
+    
+    public AtlasMapping loadMapping(String fileName, AtlasMappingFormat format) throws AtlasValidationException {
```

Comment:
I wonder whether we should use the variant with file name and InputStream to create a `Reader` and then only have a two method staking a Reader for performing the unmarshaling. This would reduce quite some duplicated code (and switch statements)